From ee161db44560495728cb0fd95eeea9041f21ce57 Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Thu, 2 May 2013 17:05:42 +0800
Subject: libstagefright: enhance error handler in MPEG4EXtractor

BZ: 92698

Enhance error handler in MPEG4Extractor, there's case that nalLength > size
but nalLength + 4 < size, for example nalLength = 0xffffffff.

Category: aosp improvement
Domain: Video.Media-container
Origin: internal
Upstream-Candidate: yes

Change-Id: Idb9de12623f9afbd5356664ebf0ee79369c3123c
Orig-Change-Id: I6036d34261ca8591679082c3a85c7e34ee42a393
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
Signed-off-by: Yuanjun Huang <yuanjun.huang@intel.com>
---
 media/libstagefright/MPEG4Extractor.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index db25516..2bdb775 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -3273,7 +3273,7 @@ status_t MPEG4Source::read(
                 if (!isMalFormed) {
                     nalLength = parseNALSize(&mSrcBuffer[srcOffset]);
                     srcOffset += mNALLengthSize;
-                    isMalFormed = srcOffset + nalLength > size;
+                    isMalFormed = nalLength > size | srcOffset + nalLength > size;
                 }
 
                 if (isMalFormed) {
-- 
1.7.4.1

